python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -478,3 +479,40 @@ def cleanup(self):
     @staticmethod
     def bootstrap_config(cluster_config):
         return bootstrap_aws(cluster_config)
+
+    @staticmethod
+    def fillout_available_node_types_resources(
+            cluster_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Fills out missing "resources" field for available_node_types."""
+        if "available_node_types" not in cluster_config:
+            return cluster_config
+        instances_list = boto3.client("ec2").describe_instance_types()[
+            "InstanceTypes"]
+        instances_dict = {
+            instance["InstanceType"]: instance
+            for instance in instances_list
+        }
+        available_node_types = cluster_config["available_node_types"]
+        for node_type in available_node_types:
+            instance_type = available_node_types[node_type]["node_config"][
+                "InstanceType"]
+            if "resources" not in available_node_types[node_type] and \
```

Comment:
Could we merge the user's resources dict with the one we fetch? 

It would be nice if a user could go

```
gpu_with_custom:
    node_config:
        InstanceType: "p3.2xlarge"
    resources: {"custom": 1} # <= fill in the rest here
```

Reply:
This is somewhat risky. Why don't we autofill only if resources was not given? What if the explicitly wanted to use only the resources.
lets say that the user had:
resources: {"CPU": 1} but the instance had 4 CPUs, we can interpret that he wanted to use only 1 CPU.

Reply:
yes; merge with a user-wins conflict policy

Reply:
and what if the user wanted to use the CPU only? 
We can share the same instance and each uses a CPU and a GPU.
I feel like what you suggest will make it "too smart"/hard to maintain/debug. @ericl can you please help break the tie here?

Reply:
Can we just do autodetected_resources.update(user_resources)? If the user wants to delete a resource they can override with "0" value.